packages/core/src/useForm/useForm.spec.ts

```diff
@@ -267,6 +267,49 @@ describe('form reset', () => {
     expect(isTouched('foo')).toBe(true);
   });
 
+  test('can reset form path values and touched to their original state', async () => {
+    const { values, reset, setValue, isTouched, setTouched } = await renderSetup(() => {
+      return useForm({
+        initialValues: { company: { employee: { name: 'John' } } },
+        initialTouched: { company: { employee: { name: true } } },
+      });
+    });
+
+    setValue('company.employee.name', 'Alice');
+    setTouched('company.employee.name', false);
+
+    expect(values).toEqual({ company: { employee: { name: 'Alice' } } });
```

Comment:
Maybe its just on my end, but I'm not sure why I'm getting these odd type errors here. The tests pass and run fine, but vscode just complains:


![CleanShot X 2025-03-21 11 57 08](https://github.com/user-attachments/assets/2b6521f4-b337-4b2e-9a28-b2c7250d7c3d)

Reply:
Yea I'm getting these too, it seems like `useForm` isn't able to infer the form type from the initial values in this case.

I will dig a little bit because it should work.

Reply:
Found the issue, there was no type connection between the initial values and the schema. I corrected it on the main branch, feel free to rebase and it should be gone.